nni/mutable/symbol.py

```diff
@@ -0,0 +1,439 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""
+Infrastructure for symbolic execution.
```

Comment:
I think it needs a brief description for "symbolic execution".
What it is used for and a minimal example about how to use it.